### PR TITLE
REL: 5.0.1

### PR DESCRIPTION
**Pull request by pubpub-zz**
### New Features (ENH)
- Add `full` parameter to PdfWriter constructor (#2865)

### Bug Fixes (BUG)
- Update pyproject.toml with minimum Python version of 3.8 (#2859)
- Cope with unbalanced delimiters in dictionary object (#2878)
- Cope with encoding with too many differences (#2873)
- Missing spaces in extract_text() method (#1328) (#2868)
- Tolerate truncated files and no warning when jumping startxref (#2855)

### Robustness (ROB)
- Repair PDF with invalid Root object (#2880)
- Continue parsing dictionary object when error is detected (#2872)
- Merge documents with invalid pages in named destinations (#2857)
- Tolerate comments in arrays (#2856)

### Developer Experience (DEV)
- Use latest Python version for benchmarking (#2879)

### Maintenance (MAINT)
- Add tests to source distributions (#2874)
- Refactor _update_field_annotation (#2862)

[Full Changelog](https://github.com/py-pdf/pypdf/compare/5.0.0...5.0.1)

### CODE REVIEW

**Comment by pubpub-zz** [comment]
@stefan6419846 
Once you have reviewed the PR, il will try to finish the process and try to sign the tag.

**Comment by pubpub-zz** [comment]
BTW, I left the raw changelog in the first commit if you wan to have a look

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+
+### Robustness (ROB)
+- Repair PDF with invalid Root object (#2880)
+- Continue parsing dictionary object when error is detected (#2872)
+- Merge documents with invalid pages in named destinations (#2857)
+- Tolerate comments(%) in arrays (#2856)
+
+### Documentation (DOC), Testing (TST), Code Style (STY), Developer Experience (DEV), Maintenance (MAINT)
+
+- (#2844), (#2862), (#2863), (#2847), (#2860), (#2867), (#2874), (#2879)
```
Comment (stefan6419846) [inline CHANGELOG.md:23]: Please do not do something like this - the grouping and listing PR numbers does not help anyone when viewing the changelog.

I personally would only include the MAINT and DEV cases here, but in their usual format.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,32 @@
...
+- Cope with unbalanced delimiters in dictionary object (#2878)
+- Cope with encoding with too many differences (#2873)
+- Missing spaces in extract_text() method (#1328) (#2868)
+- Tolerate truncated files and no warning when jumping startxref (#2855)
+
+### Robustness (ROB)
+- Repair PDF with invalid Root object (#2880)
+- Continue parsing dictionary object when error is detected (#2872)
+- Merge documents with invalid pages in named destinations (#2857)
+- Tolerate comments(%) in arrays (#2856)
```
Comment (stefan6419846) [inline CHANGELOG.md:19]:
```suggestion
- Tolerate comments in arrays (#2856)
```

**Review by stefan6419846 — APPROVED** [review]
Thanks, look good to me.

Please make sure to change the entries we have changed here accordingly where necessary to ensure a consistent changelog.

**Pull request merged**